### PR TITLE
Fix Streaming/REST Functionality and Introduce Microphone Example

### PR DESCRIPTION
**Pull request by dvonthenen**
Fixes: https://github.com/deepgram/deepgram-dotnet-sdk/issues/222, https://github.com/deepgram/deepgram-dotnet-sdk/issues/194, and https://github.com/deepgram/deepgram-dotnet-sdk/issues/180

Changes:
- Fixes breaking all functionality on `alpha.4` because of API Key issue. Tested Speak, Prerecorded and Live using Microphone.
- Introduces a Microphone example using portaudio
- Introduced `Deepgram.Microphone` and `Deepgram.Microphone.Unstable` builds on nuget
- Removed the LiveClientTests as none of the tests make sense anymore. WIll need to come up with new ones.

## Summary by CodeRabbit

- **New Features**
	- Introduced a new Deepgram Microphone project for audio input handling.
	- Implemented functionality for real-time transcription from microphone audio streams.
- **Enhancements**
	- Updated API key handling across various clients to improve security and usability.
	- Improved audio processing with the initialization and termination of PortAudio.
	- Enhanced project configurations and paths for better organization and clarity.
- **Bug Fixes**
	- Fixed nullable types in models to ensure required values are provided, enhancing data integrity.
- **Documentation**
	- Updated project references and environment variable handling in example projects for a more streamlined setup.
- **Chores**
	- Added clean-up scripts for environment maintenance.
- **Refactor**
	- Renamed projects for better alignment with their functionality.
	- Updated constructors and method signatures across clients for consistency and simplicity.

### CODE REVIEW

File: Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs [hunk 1]
```diff
@@ -56,4 +60,54 @@ public class DeepgramClientOptions
...
+        //checks for http:// https:// http https - https:// is include to ensure it is all stripped out and correctly formatted 
+        regex = new Regex(@"\b(http:\/\/|https:\/\/|http|https)\b", RegexOptions.IgnoreCase);
+        if (!regex.IsMatch(BaseAddress))
+        {
+            //if no protocol in the address then https:// is added
+            //Console.WriteLine($"REST BaseAddress: {BaseAddress}"); // TODO: logging
+            BaseAddress = $"https://{BaseAddress}";
+        }
+        BaseAddress = BaseAddress.TrimEnd('/');
+        //Console.WriteLine($"REST BaseAddress (Final): {BaseAddress}"); // TODO: logging
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs:111]: Will implement logging and fix comments later.

File: Deepgram.Tests/UnitTests/ClientTests/AnalyzeClientTests.cs [hunk 2]
```diff
@@ -16,8 +16,8 @@ public class AnalyzeClientTests
     [SetUp]
     public void Setup()
     {
-        _options = new DeepgramClientOptions();
         _apiKey = new Faker().Random.Guid().ToString();
+        _options = new DeepgramClientOptions(_apiKey);
     }
 
     [Test]
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram.Tests/UnitTests/ClientTests/AnalyzeClientTests.cs]: Will have to think about this...

File: Deepgram.Microphone/Library.cs [hunk 3]
```diff
@@ -0,0 +1,19 @@
+﻿
+namespace Deepgram.Microphone;
+
+public class Library
+{
+    public static void Initialize()
+    {
+        // TODO: logging
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram.Microphone/Library.cs:8]: Will implement logging and fix comments later.

File: Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs:111]: Will think about it.

File: Deepgram.Dev.sln [hunk 4]
```diff
@@ -7,11 +7,39 @@ Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Deepgram", "Deepgram\Deepgr
...
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "hello-world", "hello-world", "{80C7E9DD-18A8-4A24-AAF9-8BE7484A0D05}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "microphone", "microphone", "{4F2E8A7F-A634-470F-8897-F5F14DBDA2CB}"
+EndProject
+Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "file", "file", "{9A92C390-7E41-40B6-BF22-6D63B4FB6C64}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Streaming", "examples\streaming\file\Streaming.csproj", "{FD8507CC-EECF-4750-81AF-3CF8E536C007}"
+EndProject
+Project("{9A19103F-16F7-4668-BE54-9A1E7A4F7556}") = "Streaming", "examples\streaming\microphone\Streaming.csproj", "{647D3479-6725-45D0-A5F4-51D1B6E9C256}"
+EndProject
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram.Dev.sln]: Updated and removed old references.

File: Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs:111]: Will think about it.

File: Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs [hunk 5]
```diff
@@ -24,15 +31,10 @@ public class DeepgramClientOptions
     public string APIVersion { get; set; } = "v1";
 
     /// <summary>
-    /// Additional headers 
+    /// Global headers to always be added to the request
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs:32]: This should be fine now and a non-issue.

File: Deepgram/Clients/Live/v1/Client.cs [hunk 6]
```diff
@@ -63,19 +42,26 @@ public Client(string apiKey = "", DeepgramClientOptions? options = null)
...
                 _clientWebSocket.Options.SetRequestHeader(header.Key, header.Value);
             }
         }
+        if (headers is not null)
+        {
+            foreach (var header in headers)
+            {
+                _clientWebSocket.Options.SetRequestHeader(header.Key, header.Value);
+            }
+        }
```
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Clients/Live/v1/Client.cs:64]: That's intentional. It isnt always needed.

File: Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs (hunk 1 above)
[the remark this answers is not in the document]
Comment (dvonthenen) [reply Deepgram/Models/Authenticate/v1/DeepgramClientOptions.cs:111]: I think it's too complex to push it into another function. That's what cause my initial "oops" in the first place.

**Review by lukeocodes — APPROVED** [review]

**Pull request merged**